code/modules/admin/admin.dm

```diff
@@ -25,6 +25,68 @@ var/global/floorIsLava = 0
 /datum/admins/SDQL_update(var/const/var_name, var/new_value)
 	return 0
 
+/datum/admins/proc/newban(var/mob/M)
```

Comment:
As a final issue with this (because who doesn't love protracted review:
There's nothing stopping a mob ghosting during these inputs.
I'd rather this took a client directly, which to_chat will still work for, stored the ckey, and then checked if that client still existed after the inputs for messages, which stops a mob ghosting or logging out to dodge a bannu.